rai/node/node.cpp

```diff
@@ -2345,6 +2479,69 @@ bool rai::block_arrival::recent (rai::block_hash const & hash_a)
 	return arrival.get<1> ().find (hash_a) != arrival.get<1> ().end ();
 }
 
+rai::online_reps::online_reps (rai::node & node) :
+node (node)
+{
+}
+
+void rai::online_reps::vote (std::shared_ptr<rai::vote> const & vote_a)
+{
+	auto rep (vote_a->account);
+	std::lock_guard<std::mutex> lock (mutex);
+	auto now (std::chrono::steady_clock::now ());
+	while (!reps.empty () && reps.begin ()->last_heard + std::chrono::seconds (rai::node::cutoff) < now)
+	{
+		auto it (reps.begin ());
+		auto old_stake (online_stake);
+		online_stake -= node.weight (it->representative);
+		if (online_stake > old_stake)
+		{
+			// overflow
+			online_stake = 0;
+		}
+		reps.erase (it);
+	}
+	auto rep_it (reps.get<1> ().find (rep));
+	auto info (rai::rep_last_heard_info{ now, rep });
+	if (rep_it == reps.get<1> ().end ())
+	{
+		auto old_stake (online_stake);
+		online_stake += node.weight (rep);
+		if (online_stake < old_stake)
+		{
+			// overflow
+			online_stake = ~0;
```

Comment:
While correct, `std::numeric_limits<rai::uint128_t>::max()` is in line with the rest of the code base and makes the semantics clear.

Reply:
I think this also needs to be done because the negation is being done on an int type before conversion to the uint128_t

Reply:
The integral conversion rules makes it work (~0 is signed). 

But ::max is clearer.